kratos/processes/levelset_convection_process.h

```diff
@@ -225,19 +248,76 @@ class LevelSetConvectionProcess
 
             // Emulate clone time step by copying the new distance onto the old one
-            #pragma omp parallel for
-            for (int i_node = 0; i_node < static_cast<int>(mpDistanceModelPart->NumberOfNodes()); ++i_node){
-                auto it_node = mpDistanceModelPart->NodesBegin() + i_node;
+            if (mBfeccOrder == 0){
+                #pragma omp parallel for
+                for (int i_node = 0; i_node < static_cast<int>(mpDistanceModelPart->NumberOfNodes()); ++i_node){
+                    auto it_node = mpDistanceModelPart->NodesBegin() + i_node;
 
-                const array_1d<double,3>& v = mVelocity[i_node];
-                const array_1d<double,3>& v_old = mVelocityOld[i_node];
+                    const array_1d<double,3>& v = mVelocity[i_node];
+                    const array_1d<double,3>& v_old = mVelocityOld[i_node];
 
-                it_node->FastGetSolutionStepValue(VELOCITY) = Nold * v_old + Nnew * v;
-                it_node->FastGetSolutionStepValue(VELOCITY, 1) = Nold_before * v_old + Nnew_before * v;
-                it_node->FastGetSolutionStepValue(mrLevelSetVar, 1) = it_node->FastGetSolutionStepValue(mrLevelSetVar);
+                    it_node->FastGetSolutionStepValue(mrConvectVar) = Nold * v_old + Nnew * v;
+                    it_node->FastGetSolutionStepValue(mrConvectVar, 1) = Nold_before * v_old + Nnew_before * v;
+                    it_node->FastGetSolutionStepValue(mrLevelSetVar, 1) = it_node->FastGetSolutionStepValue(mrLevelSetVar);
+                }
             }
+            else{
+                #pragma omp parallel for
+                for (int i_node = 0; i_node < static_cast<int>(mpDistanceModelPart->NumberOfNodes()); ++i_node){
+                    auto it_node = mpDistanceModelPart->NodesBegin() + i_node;
+
+                    const array_1d<double,3>& v = mVelocity[i_node];
+                    const array_1d<double,3>& v_old = mVelocityOld[i_node];
+
+                    it_node->FastGetSolutionStepValue(mrConvectVar) = Nold * v_old + Nnew * v;
+                    it_node->FastGetSolutionStepValue(mrConvectVar, 1) = Nold_before * v_old + Nnew_before * v;
+                    const double dist = it_node->FastGetSolutionStepValue(mrLevelSetVar);
+                    it_node->FastGetSolutionStepValue(mrLevelSetVar, 1) = dist;
+                    it_node->SetValue(mrLevelSetVar, dist);
+                }
+            }
+
+            mpSolvingStrategy->Solve(); // forward convection to reach phi_n+1
+
+            if (mBfeccOrder > 0) {// Error Compensation and Correction
+                #pragma omp parallel for
+                for (int i_node = 0; i_node < static_cast<int>(mpDistanceModelPart->NumberOfNodes()); ++i_node){
+                    auto it_node = mpDistanceModelPart->NodesBegin() + i_node;
+
+                    const array_1d<double,3>& v = mVelocity[i_node];
+                    const array_1d<double,3>& v_old = mVelocityOld[i_node];
```

Comment:
```suggestion
                    const array_1d<double,3>& r_v = mVelocity[i_node];
                    const array_1d<double,3>& r_v_old = mVelocityOld[i_node];
```
Style guide (same in lines above).